dateparser/conf.py

```diff
@@ -79,8 +81,135 @@ def wrapper(*args, **kwargs):
             kwargs['settings'] = settings.replace(mod_settings=mod_settings, **kwargs['settings'])
 
         if not isinstance(kwargs['settings'], Settings):
-            raise TypeError(
-                "settings can only be either dict or instance of Settings class")
+            raise TypeError("settings can only be either dict or instance of Settings class")
 
         return f(*args, **kwargs)
     return wrapper
+
+
+class SettingValidationError(ValueError):
+    pass
+
+
+def _check_require_part(provided_list):
+    """Returns `True` if the provided list of parts is valid"""
+    return not set(provided_list) - {'day', 'month', 'year'}
```

Comment:
should I check the length to avoid duplicated parsers?

Reply:
I think that would be great.